s3same/iam.py

```diff
@@ -0,0 +1,75 @@
+from botocore.exceptions import ClientError
+
+IAMName = 's3same_travis'
+
+def _policy_string(bucket):
+    return """{
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Action": [
+            "s3:ListBucket"
+          ],
+          "Effect": "Allow",
+          "Resource": [
+            "arn:aws:s3:::%(bucket)s"
+          ]
+        },
+        {
+          "Action": [
+            "s3:PutObject",
+            "s3:PutObjectAcl"
+          ],
+          "Effect": "Allow",
+          "Resource": [
+            "arn:aws:s3:::%(bucket)s/*"
+          ]
+        }
+      ]
+    }""" % dict(bucket=bucket)
+
+def _policy_arn(iam, bucket):
+    policy_arn = None
+    try:
+        response = iam.create_policy(
+                PolicyName=IAMName,
+                PolicyDocument=_policy_string(bucket),
+                )
+        policy_arn = response.get('Policy', {}).get('Arn')
+    except ClientError as e:
+        if e.response['Error']['Code'] != 'EntityAlreadyExists':
+            raise e
+        marker = None
+        while not policy_arn:
+            response = iam.list_policies(Scope='Local')
+            policy_arn = next((p.get('Arn') for p in response.get('Policies', []) if p.get('PolicyName') == IAMName), None)
+            if not response.get('IsTruncated'):
+                break
+            marker = response['Marker']
```

Comment:
This might just be the 1am fuzziness, but I'm thinking a more pythonic way might be to write a generator-function that retrieves all entities at a paged endpoint (do a request, yield those one at a time, then when they run out, if there's more, get the next page, keep yielding, etc.).  With that made, this code would be the `next`-wrapped generator comprehension on the retrieve-all generator function, so much simpler.